std/assembly/internal/typedarray.ts

```diff
@@ -123,4 +123,29 @@ export abstract class TypedArray<T,V> {
       return this;
     }
   }
+
+  map<W extends TypedArray<T, V>>(callbackfn: (value: T, index: i32, array: W) => T): TypedArray<T, V> {
+    var buffer = this.buffer;
+    var length = this.length;
+    var byteLength = this.byteLength;
+    var byteOffset = this.byteOffset;
+    var newbuff = allocateUnsafe(byteLength);
```

Comment:
Why just not create `var result = new TypedArray<T,V>(this.length)`?

Reply:
It's in the definition of the class

```ts
export abstract class TypedArray<T,V>
```

Reply:
yeah, now I see

Reply:
`TypedArray` is abstract because you'd lose the `subarray` implementation on the concrete class. Though I agree that this should be easier, somehow, because as it stands it is missing GC registration.

Reply:
Oh wait! I have an idea.

Reply:
Maybe a built-in helper `instantiate<T>(...constructorArgs)` that is able to instantiate anything, given the class as a type argument?

Reply:
@dcodeIO I can implement the method on each subclass using the `new` keyword. Let me submit changes.